app/main.py

```diff
@@ -1,2 +1,92 @@
+from typing import Any, Hashable, Iterable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.table: list = [None] * 8
+        self.__capacity = len(self.table)
+
+    def __len__(self) -> int:
+        return self.length
+
+    def clear(self) -> None:
+        self.table = [None] * self.__capacity
+        self.length = 0
+
+    def __delitem__(self, key: Hashable) -> None:
+        hashed_key = hash(key)
+        index = hashed_key % self.__capacity
+
+        if self.table[index] is not None:
+            for i, (k, h, v) in enumerate(self.table[index]):
+                if k == key:
+                    del self.table[index][i]
+                    self.length -= 1
+                    return
+
+        raise KeyError(key)
```

Comment:
Imho `key` it is not very descriptive error message : ( Fix that please

Reply:
ok, ty. I try to fix that)